src/SSVOpenHexagon/Core/HGUpdate.cpp

```diff
@@ -51,15 +51,22 @@ void HexagonGame::update(ssvu::FT mFT)
     const std::string diffStr = diffFormat(difficultyMult);
     const std::string timeStr = timeFormat(status.getTimeSeconds());
 
+    constexpr float DELAY_TO_UPDATE = 5.f; // X seconds
+    timeUntilRichPresenceUpdate -= ssvu::getFTToSeconds(mFT);
+
     // Presence formatter
-    const std::string presenceStr =
-        nameStr + " [x" + diffStr + "] - " + timeStr + "s";
+    const std::string presenceStr = nameStr + " [x" + diffStr + "] - " + timeStr + "s";
+
+    if (timeUntilRichPresenceUpdate <= 0.f)
+    {
+        steamManager.set_rich_presence_in_game(nameStr, diffStr, timeStr);
+        steamManager.run_callbacks();
```

Comment:
`run_callbacks()` should be executed per frame, both here and for `discordManager`